consensus/state.go

```diff
@@ -837,11 +1139,12 @@ func (cs *ConsensusState) FinalizeCommit(height int) {
 	log.Info(Fmt("Finalizing commit of block: %v", cs.ProposalBlock))
 	// We have the block, so stage/save/commit-vote.
 	cs.saveBlock(cs.ProposalBlock, cs.ProposalBlockParts, cs.Votes.Precommits(cs.CommitRound))
-	// Increment height.
-	cs.updateToState(cs.stagedState)
+
+	// call updateToState from handleTimeout
```

Comment:
We should increment the height here and now because we have a commit, otherwise we're introducing unnecessary latency from time of commit to start of round0(@height+1).

So that's why the spec is slightly weird.